src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -1278,6 +1289,7 @@ private function createSubCrawler($nodes)
         $crawler->isHtml = $this->isHtml;
         $crawler->document = $this->document;
         $crawler->namespaces = $this->namespaces;
+        $crawler->cachedNamespaces = $this->cachedNamespaces;
```

Comment:
give a try to `$crawler->cachedNamespaces = &$this->cachedNamespaces;` to share the same cache as suggested in https://github.com/symfony/symfony/pull/39097#issuecomment-730349079

Reply:
Good idea too. It's just not very intransparent and tricky to allow for example a subcrawler to clear the cache and build a new one without affecting the parent crawlers. Works out of the box with an ArrayIterator.